doc generation should skip rector-recipe.php

In `rector-recipe.php` I used `SetList::DOWNGRADE`. 
This constant was removed but it is still there in my rector-recipe.php, this causes an error in the console output:

> Script bin/rector dump-rectors > docs/rector_rules_overview.md handling the docs event returned with error code 1

and in the `rector_rules_overview.md:
> [ERROR] Undefined class constant 'DOWNGRADE'

--- a/src/Bootstrap/RectorConfigsResolver.php
+++ b/src/Bootstrap/RectorConfigsResolver.php
@@ -81,10 +81,12 @@ final class RectorConfigsResolver
 
         $setFileInfos = $this->resolveSetFileInfosFromConfigFileInfos($configFileInfos);
 
-        // autoload rector recipe file if present
-        $rectorRecipeFilePath = getcwd() . '/rector-recipe.php';
-        if (file_exists($rectorRecipeFilePath)) {
-            $configFileInfos[] = new SmartFileInfo($rectorRecipeFilePath);
+        if (in_array($argvInput->getFirstArgument(), ['generate', 'g', 'create', 'c'], true)) {
+            // autoload rector recipe file if present, just for \Rector\RectorGenerator\Command\GenerateCommand
+            $rectorRecipeFilePath = getcwd() . '/rector-recipe.php';
+            if (file_exists($rectorRecipeFilePath)) {
+                $configFileInfos[] = new SmartFileInfo($rectorRecipeFilePath);
+            }
         }
 
         return array_merge($configFileInfos, $setFileInfos);

